src/internal/operations/restore.go

```diff
@@ -227,7 +227,11 @@ func (op *RestoreOperation) do(
 		return nil, clues.Wrap(err, "getting backup and details")
 	}
 
-	restoreToProtectedResource, err := chooseRestoreResource(ctx, op.rc, op.RestoreCfg, bup.Selector)
+	restoreService, restoreToProtectedResource, err := op.rc.GetRestoreResource(
```

Comment:
is there a way the selector and this service can differ? Could we remove the service return value and continue to use the value from the selector? Seems like it would be less error prone